workflow/artifacts/s3/s3.go

```diff
@@ -103,6 +104,43 @@ func loadS3Artifact(s3cli argos3.S3Client, inputArtifact *wfv1.Artifact, path st
 	return true, nil
 }
 
+// OpenStream opens a stream reader for an artifact from S3 compliant storage
+func (s3Driver *ArtifactDriver) OpenStream(inputArtifact *wfv1.Artifact) (io.ReadCloser, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	log.Infof("S3 OpenStream: key: %s", inputArtifact.S3.Key)
+	s3cli, err := s3Driver.newS3Client(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new S3 client: %v", err)
+	}
+
+	return streamS3Artifact(s3cli, inputArtifact)
+
+}
+
+func streamS3Artifact(s3cli argos3.S3Client, inputArtifact *wfv1.Artifact) (io.ReadCloser, error) {
+	stream, origErr := s3cli.OpenFile(inputArtifact.S3.Bucket, inputArtifact.S3.Key)
```

Comment:
minor - but `origErr` could just be `err` I think

Reply:
It's used on line 137 after err is modified